network_wrangler/RoadwayNetwork.py

```diff
@@ -118,41 +118,57 @@ def write(self, filename: str, path: str = '.') -> None:
         shapes_file = os.path.join(path, filename + "_shape.geojson")
         self.shapes_df.to_file(shapes_file, driver='GeoJSON')
 
-
-
-    def apply_roadway_feauture_change(self, project_card: str) -> bool:
+    def apply_roadway_feauture_change(self, card_dict: dict) -> bool:
         '''
-        Changes the road network according to the project card passed
-        
+        Changes the road network according to the project card information passed
+
         args:
-        project_card: the path to the project card
-        
+        card_dict: dictionary with project card information
+
         returns:
         bool: True if successful.
         '''
-        #import pdb; pdb.set_trace()
-        with open(project_card, 'r') as card:
-            card_dict = yaml.safe_load(card)
-            
-            road_dict = card_dict.get("Road")            
-            name_str = road_dict.get(list(road_dict)[0])
-            id = name_str.split('=')[1]
-            
-            change_dict = card_dict.get("Change")
-            existing_value = change_dict.get(list(change_dict)[0])
-            build_value = change_dict.get(list(change_dict)[1])
-            
-            attribute = card_dict.get("Attribute")
-            
-            print(id)
-            print(attribute)
-            print(existing_value)
-            print(build_value)
-            
-            
-            #check existing value that it matches the attribute field in self.links_df
-            #TODO: how do i manipulate the data in the dataframe to apply the changes?
-            
-            self.links_df
-            
+
+        road_dict = card_dict.get("Road")
+        road_id = road_dict.get("Name").split("=")[1]
+
+        attribute = card_dict.get("Attribute").upper()
```

Comment:
typically you would access a dictionary value by using dict['key'] rather than dict.get()
The dict.get() method is slower, but is useful if you want to provide a default value in case it doesn't exist: i.e. 

    my_dict = {'A':1, 'B':2}
    my_dict.get('C', 'not here')

    # returns 'not here'

Reply:
In general though, you probably don't need to create separate local variables for most of these.